src/Text/Pandoc/Writers/RST.hs

```diff
@@ -353,9 +353,24 @@ blockListToRST' :: PandocMonad m
                 -> [Block]       -- ^ List of block elements
                 -> RST m Doc
 blockListToRST' topLevel blocks = do
+  let transformLigature b1 b2@(BlockQuote _)
+         | isListBlock b1 = [b1, commentSep, b2]
+         | otherwise      = [b1, b2]
+        where
+          isListBlock (BulletList _)     = True
+          isListBlock (OrderedList _ _)  = True
+          isListBlock (DefinitionList _) = True
+          isListBlock _                  = False
+          commentSep  = RawBlock "rst" ""
+      transformLigature b1 b2 = [b1, b2]
```

Comment:
This way of writing `transformLigature` causes a new list to be built up for every two elements -- that seems inefficient.

Also, what if the sequence is: BlockQuote, List, BlockQuote? If I understand the logic, `fixBlocks` won't insert a comment between the second and third elements in this case...

Why not something simpler like:
```
fixBlocks (b1:b2:bs)
  | isList b1 && isBlockQuote b2 = b1 : commentsep : b2 : fixBlock bs
fixBlocks (b:bs) = b : fixBlocks bs
fixBlocks [] = []
```

Reply:
> Also, what if the sequence is: BlockQuote, List, BlockQuote? If I understand the logic, fixBlocks won't insert a comment between the second and third elements in this case...

yes you are right i didn't do a good job with pattern-matching there.

> Why not something simpler like ...

cool i updated to adopt your version. `transformLigature` was an attempt at generalising `fixBlocks`. i am not familiar with the use of a single guard in a function body, and i didn't find any reference about its usage online. what happens when the condition is not met? i assume that the next pattern match (at the function signature level) is evaluated.

let me know whether anything else is missing